cmd/inspect.go

```diff
@@ -27,80 +27,123 @@ import (
+func getInspectCmd(logger *logrus.Logger) *cobra.Command {
+	var addExecReqs bool
+
 	// inspectCmd represents the inspect command
 	inspectCmd := &cobra.Command{
 		Use:   "inspect [file]",
 		Short: "Inspect a script or archive",
 		Long:  `Inspect a script or archive.`,
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			pwd, err := os.Getwd()
-			if err != nil {
-				return err
-			}
-			filesystems := loader.CreateFilesystems()
-			src, err := loader.ReadSource(logger, args[0], pwd, filesystems, os.Stdin)
+			src, filesystems, err := readSource(args[0], logger)
 			if err != nil {
 				return err
 			}
 
-			typ := runType
-			if typ == "" {
-				typ = detectType(src.Data)
-			}
-
 			runtimeOptions, err := getRuntimeOptions(cmd.Flags(), buildEnvMap(os.Environ()))
 			if err != nil {
 				return err
 			}
 			registry := metrics.NewRegistry()
-			_ = metrics.RegisterBuiltinMetrics(registry)
+			builtinMetrics := metrics.RegisterBuiltinMetrics(registry)
 
-			var (
-				opts lib.Options
-				b    *js.Bundle
-			)
-			switch typ {
+			var b *js.Bundle
+			switch getRunType(src) {
+			// this is an exhaustive list
 			case typeArchive:
 				var arc *lib.Archive
 				arc, err = lib.ReadArchive(bytes.NewBuffer(src.Data))
 				if err != nil {
 					return err
 				}
 				b, err = js.NewBundleFromArchive(logger, arc, runtimeOptions, registry)
-				if err != nil {
-					return err
-				}
-				opts = b.Options
+
 			case typeJS:
 				b, err = js.NewBundle(logger, src, filesystems, runtimeOptions, registry)
+			}
+			if err != nil {
+				return err
+			}
+
+			// ATM, output can take 2 forms: standard (equal to lib.Options struct) and extended, with additional fields.
+			inspectOutput := interface{}(b.Options)
+
+			if addExecReqs {
+				inspectOutput, err = addExecRequirements(b, builtinMetrics, registry, logger)
 				if err != nil {
 					return err
 				}
-				opts = b.Options
 			}
 
-			data, err := json.MarshalIndent(opts, "", "  ")
+			data, err := json.MarshalIndent(inspectOutput, "", "  ")
 			if err != nil {
 				return err
 			}
 			fmt.Println(string(data))
+
 			return nil
 		},
 	}
 
 	inspectCmd.Flags().SortFlags = false
 	inspectCmd.Flags().AddFlagSet(runtimeOptionFlagSet(false))
 	inspectCmd.Flags().StringVarP(&runType, "type", "t", runType, "override file `type`, \"js\" or \"archive\"")
+	inspectCmd.Flags().BoolVar(&addExecReqs,
+		"execution-requirements",
+		false,
+		"include calculations of execution requirements for the test")
 
 	return inspectCmd
 }
+
+func addExecRequirements(b *js.Bundle,
+	builtinMetrics *metrics.BuiltinMetrics, registry *metrics.Registry,
+	logger *logrus.Logger) (interface{}, error) {
+	runner, err := js.NewFromBundle(logger, b, builtinMetrics, registry)
+	if err != nil {
+		return nil, err
+	}
+
+	conf, err := getConsolidatedConfig(afero.NewOsFs(), Config{}, runner)
+	if err != nil {
+		return nil, err
```

Comment:
Bundle is actually a type that I and @na-- have mostly wanted to remove in the past, so exposing it more seems like a bad idea. Refactoring everything needed is probably going to be way too much work, but I would argue you that given that you already got the Bundle, and that it actually *has* the options and that `getConsolidatedConfig` just gets the Options from the runner, which gets it from the Bundle inside .... You can just rewrite getConsolidatedConfig to be taking the Options and get the Options directly from the runner, skipping making `NewFromBundle` public.  

Even more importantly the one thing that `get

Reply:
I suppose you meant this https://github.com/grafana/k6/issues/1048 ?

Reply:
I think that issue (as most very old issues) is now very outdated, but in parts - yes.

Reply:
@MStoykov  I've looked into this: trying to refactor in "minimal fashion" (without touching `js/` stuff) gets stuck on the point of `Runner.IsExecutable` → which needs `Bundle.exports` so again, the same problem with private field of `Bundle`. Trying to refactor this at the level of `Bundle` is definitely too out-of-scope for this PR: I think it should be done separately. 
For now, I'll add some notes on that to the #1048 issue: it can be continued in a separate PR or be left alone until next prioritization.

I added a small refactoring around `getConsolidatedConfig`: it can truly be used with only options, without `Runner`: I think it's relatively small to be present in this PR and might simplify next stages of refactorings too.

Reply:
:( 

Hm looking at this again, maybe just instead of using NewBundle and then NewFromBundle, you can just go with creating the runner the same `run` will do. The additional work `NewFromBundle` does is ... minimal and should not have any side effects. But will make the cmd<->js dependency smaller. 

WDYT ?

Reply:
I wouldn't say dependency would be "smaller".. meaning, even prior to this PR `inspect` was the only one in cmd using `Bundle` directly which makes sense in its case. From this follows that probably any major refactoring in `js` will impact `inspect` first.

Of course, it can be re-written as you describe but TBH, I think all options here are equally bad :joy: -- because that major refactoring is needed. I slightly prefer the current version because I don't see any harm from public `NewFromBundle` at this point while this version more explicitly shows the use cases that future refactoring should take into account. So it's kind of a way to document too.